openrarity/collection/collection.py

```diff
@@ -32,27 +32,73 @@
 
 
 class AttributesStatistics(TypedDict):
+    """
+    A class to represent Attribute Statistic.
+    AttributeStatistic :
+        Attribute Statistics can be calculated by grouping `name` and `value` attributes.
+        Attribute Statistics include
+            - attribute.token_count
+            - attribute.supply
+            - metric.probability
+            - metric.information
+        For definitions, Please take a look at `rank_collection()` method.
+    """
     string: list[AttributeStatistic]
     number: list[AttributeStatistic]
     date: list[AttributeStatistic]
 
 
 class TokenCollection:
-    """ """
+    """
+    The TokenCollection class is the entrypoint for the majority of OpenRarity's functionality and handles all the parsing, validation, and ranking of the tokens.
+    """
 
     def __init__(
         self,
         token_type: Literal["non-fungible", "semi-fungible"],
         tokens: dict[TokenId, RawToken],
     ):
+        """
+        Constructs all the necessary attributes for the `TokenCollection` object.
+
+        Parameters
+        ----------
+        token_type : Literal["non-fungible", "semi-fungible"]
+            Type of the token.
+        tokens : dict[TokenId, RawToken]
+            A dictionary of raw token data.
+
+        Attributes
+        ----------
+        _token_type : Literal["non-fungible", "semi-fungible"]
+            Type of the token.
+        _input_checksum : str
+            MD5 hash of the raw token data.
+        _ranks_checksum : str
+            MD5 hash of ranks data.
+        _token_supply : int | dict[str | int, int]
+            Token supply value.
+            Non-Fungible is the number of tokens in the collection where each token is unique.
+            Semi-Fungible token_supply value is a dict of token_ids with their token_supply value.
+        _tokens : dict[TokenId, RawToken]
+            A dictionary of tokens data.
+        _attribute_statistics : AttributesStatistics
+            Statistics of a given collection grouped by `name` and `value` attributes.
+        _entropy : float | None
+            Entropy value. Entropy is a measure of information in terms of uncertainity.
+        _token_statistics : list[TokenStatistic]
+            Statistics of each token in a given collection grouped by `name` and `value` attributes.
+        _ranks : list[RankedToken]
+            Rarity ranks of each token.
```

Comment:
As per the previous review comments. Don't document private attributes. Only document the public ones and properties.